src/lava/utils/profiler.py

```diff
@@ -12,3 +12,125 @@
 elementary operation has a defined execution time and energy cost, which is
 used in a performance model to calculate execution time and energy.
 """
+import typing as ty
+from abc import ABC
+import types
+import numpy as np
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.run_conditions import AbstractRunCondition
+from lava.magma.core.run_configs import RunConfig
+from lava.magma.runtime.runtime import Runtime
+from lava.magma.core.process.message_interface_enum import ActorType
+from lava.magma.compiler.compiler import Compiler
+from lava.magma.core.resources import (
+    AbstractComputeResource, Loihi1NeuroCore, Loihi2NeuroCore)
+
+
+class LavaProfiler(ABC):
+    """TBD"""
+
+    def __init__(self, start: int = 0, end: int = 0,
+                 bin_size: int = 1, buffer_size: int = 1000):
+        self.start = start
+        self.end = end
+        self.bin_size = bin_size
+        self.buffer_size = buffer_size
+        self.used_resources: ty.List[AbstractComputeResource] = []
+
+    def profile(self, proc: AbstractProcess):
+        proc.run = types.MethodType(self.run, proc)
+
+    def get_energy(self) -> np.array:
+        """Returns the energy estimate per time step in µJ."""
+        ...
+        return 0
+
```

Comment:
Should we give the user the option to either receive the whole time series or just the total time / energy? I could imagine that for particularly long runs, we run into memory or runtime problems if we store one value for each time step. If we just accumulate the values, it may often suffice.
In addition, what's about a value to provide the frequency of measurements? We may often not need a value each time step, but only each 1000th time step.

Reply:
Sure, this is just a first example of getting results. We can offer more sophisticated options, including standard plots etc.